packages/flutter/test/widget/image_test.dart

```diff
@@ -0,0 +1,220 @@
+        new NetworkImage(
+          key: new GlobalKey(),
+          scale: 1.0,
+          src: testUrl
+        )
+      );
+
+      ImageResource imageResource = imageCache.load(testUrl, scale: 1.0);
+      expect(tester.findElementByKey(new ObjectKey(imageResource)), isNull);
+    });
+  });
+
+  test('Verify AsyncImage sets an ObjectKey on it\'s ImageResource if it doesn\'t have a key', () {
+    testWidgets((WidgetTester tester) {
+      ImageProvider imageProvider = new TestImageProvider();
+      tester.pumpWidget(new AsyncImage(provider: imageProvider));
+
+      ImageResource imageResource = imageCache.loadProvider(imageProvider);
+      expect(tester.findElementByKey(new ObjectKey(imageResource)), isNotNull);
+    });
+  });
+
+  test('Verify AsyncImage doesn\'t set an ObjectKey on it\'s ImageResource if it has a key', () {
+    testWidgets((WidgetTester tester) {
+      ImageProvider imageProvider = new TestImageProvider();
+      tester.pumpWidget(
+        new AsyncImage(
+          key: new GlobalKey(),
+          provider: imageProvider
+        )
+      );
+
+      ImageResource imageResource = imageCache.loadProvider(imageProvider);
+      expect(tester.findElementByKey(new ObjectKey(imageResource)), isNull);
+    });
+  });
+
+  test('Verify AssetImage sets an ObjectKey on it\'s ImageResource if it doesn\'t have a key', () {
+    testWidgets((WidgetTester tester) {
+      final String name = 'foo';
+      final AssetBundle assetBundle = new TestAssetBundle();
+      tester.pumpWidget(
+        new AssetImage(
+          name: name,
+          bundle: assetBundle
+        )
+      );
+
+      ImageResource imageResource = assetBundle.loadImage(name);
+      expect(tester.findElementByKey(new ObjectKey(imageResource)), isNotNull);
+    });
+  });
+
+  test('Verify AssetImage doesn\'t set an ObjectKey on it\'s ImageResource if it has a key', () {
+    testWidgets((WidgetTester tester) {
+      final String name = 'foo';
+      final AssetBundle assetBundle = new TestAssetBundle();
+      tester.pumpWidget(
+        new AssetImage(
+          key: new GlobalKey(),
+          name: name,
+          bundle: assetBundle
+        )
+      );
+
+      ImageResource imageResource = assetBundle.loadImage(name);
+      expect(tester.findElementByKey(new ObjectKey(imageResource)), isNull);
+    });
+  });
+
+  test('Verify AsyncImage resets its RenderImage when changing providers if it doesn\'t have a key', () {
+    testWidgets((WidgetTester tester) {
+      final GlobalKey key = new GlobalKey();
+      TestImageProvider imageProvider1 = new TestImageProvider();
+      tester.pumpWidget(
+        new Container(
+          key: key,
+          child: new AsyncImage(
+            provider: imageProvider1
+          )
+        )
+      );
+      RenderImage renderImage = key.currentContext.findRenderObject();
+      expect(renderImage.image, isNull);
+
+      // An exception will be thrown when we try to draw the image.  Catch it.
```

Comment:
ah, it's an argument to pumpWidgets. Feel free to move the logic to pump() so you can use it just with pump.